### ISSUE TITLE
'FacetedSearchView' object has no attribute 'results'

### ISSUE THREAD

**Issue by wengole** [django-haystack/django-haystack#1159]
Just trying to follow the example in the [docs](http://django-haystack.readthedocs.org/en/latest/views_and_forms.html#views), but getting:
```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/search?q=%2A&search=search&page=3

Django Version: 1.7.7
Python Version: 2.7.9
Installed Applications:
('snapshots',
 'suit',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'crispy_forms',
 'menu',
 'haystack',
 'celery_haystack',
 'bootstrap_pagination',
 'debug_toolbar')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'debug_toolbar.middleware.DebugToolbarMiddleware')


Traceback:
File "/home/benc/buildout-cache/eggs/Django-1.7.7-py2.7.egg/django/core/handlers/base.py" in get_response
  111.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/home/benc/buildout-cache/eggs/Django-1.7.7-py2.7.egg/django/views/generic/base.py" in view
  69.             return self.dispatch(request, *args, **kwargs)
File "/home/benc/buildout-cache/eggs/Django-1.7.7-py2.7.egg/django/views/generic/base.py" in dispatch
  87.         return handler(request, *args, **kwargs)
File "/home/benc/buildout-cache/eggs/django_haystack-2.4.0.dev0-py2.7.egg/haystack/generic_views.py" in get
  118.             return self.form_invalid(form)
File "/home/benc/buildout-cache/eggs/django_haystack-2.4.0.dev0-py2.7.egg/haystack/generic_views.py" in form_invalid
  71.             'object_list': self.get_queryset()
File "/home/benc/buildout-cache/eggs/django_haystack-2.4.0.dev0-py2.7.egg/haystack/generic_views.py" in get_context_data
  101.         context.update({'facets': self.results.facet_counts()})

Exception Type: AttributeError at /search
Exception Value: 'FacetedSearchView' object has no attribute 'results'
```

**Comment by ephes** [comment]
I ran into the same issue today. There seem to be at least two bugs in haystack.generic_views.FacetedSearchMixin:

 1. FacetedSearchMixin was copied from SearchMixin and calls super(SearchMixin, self) instead of super(FacetedSearchMixin, self) in get_form_kwargs. Therefore SearchMixin.get_form_kwargs does not get called and your form does not get the query data and form.is_bound is False.
 2. In .get_context_data the call self.results.facet_counts() should be self.queryset.facet_counts().

There's a pull request for those bugs #1149. I also couldn't find any tests for haystack.generic_views. Since FacetedSearchMixin does not have that much code, I think I'm going to use my own implementation.

**Comment by narrowfail** [comment]
Version 2.4.0 seems unpached.

**Comment by chelis** [comment]
Is this fix going to be released somewhere in the near future? I am running into the same issue.

**Comment by SalahAdDin** [comment]
Here too.

**Comment by MichaelAnckaert** [comment]
Can we get the pull request referenced above merged please? The generic view FacetedSearchView is essentially useless at this point.

**Comment by SalahAdDin** [comment]
Why @MichaelAnckaert ?

**Comment by SteveByerly** [comment]
Fixed with #1149 and merged with 94558b3
